src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -33,42 +38,689 @@ const ActorInit Object_Kankyo_InitVars = {
+        spC4.y = this->unk_14C[i].unk_04 + this->unk_14C[i].unk_10;
+        spC4.z = this->unk_14C[i].unk_08 + this->unk_14C[i].unk_14;
+        func_80169474(globalCtx, &spC4, &spB8);
+
+        if ((spB8.x >= 0.0f) && (spB8.x < 320.0f) && (spB8.y >= 0.0f) && (spB8.y < 240.0f)) {
+            if (!spB4) {
+                spB4 = true;
+
+                gDPPipeSync(POLY_XLU_DISP++);
+                gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 255, 255);
+                gSPClearGeometryMode(POLY_XLU_DISP++, G_LIGHTING);
+
+                POLY_XLU_DISP = Gfx_CallSetupDL(POLY_XLU_DISP, 0);
+
+                gDPSetRenderMode(POLY_XLU_DISP++, G_RM_FOG_SHADE_A, G_RM_ZB_CLD_SURF2);
+                gSPSetGeometryMode(POLY_XLU_DISP++, G_FOG);
+                gSPSegment(POLY_XLU_DISP++, 0x08, Lib_SegmentedToVirtual(D_0408EBE0));
+            }
+
+            Matrix_InsertTranslation(spC4.x, spC4.y, spC4.z, MTXMODE_NEW);
+            tempf = (i & 7) * 0.008f;
+            Matrix_Scale(0.05f + tempf, 0.05f + tempf, 0.05f + tempf, MTXMODE_APPLY);
+            temp_f2 = Math_Vec3f_DistXYZ(&spC4, &globalCtx->view.eye) / 300.0f;
+            temp_f2 = ((1.0f < temp_f2) ? 0.0f : (((1.0f - temp_f2) > 1.0f) ? 1.0f : 1.0f - temp_f2));
+
+            gDPPipeSync(POLY_XLU_DISP++);
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, (u8)(160.0f * temp_f2));
+
+            Matrix_InsertMatrix(&globalCtx->mf_187FC, MTXMODE_APPLY);
+
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, &D_04023130);
+        }
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_808DD970(Actor* thisx, GlobalContext* globalCtx2) {
+    f32 temp_f0;
+    f32 temp_f20;
+    Vec3f spBC;
+    Vec3f spB0;
+    f32 tempf;
+    s16 i;
+    f32 phi_f26;
+    GlobalContext* globalCtx = globalCtx2;
+    ObjectKankyo* this = THIS;
+    f32 tempA;
+
+    if (globalCtx->sceneNum == SCENE_KYOJINNOMA) { // Nice
+        phi_f26 = 1.0f;
+    } else {
+        tempA = func_800E031C(GET_ACTIVE_CAM(globalCtx));
+        if (tempA != BGCHECK_Y_MIN) {
+            tempA -= globalCtx->view.eye.y;
+            phi_f26 = tempA / 4000.0f;
+        } else {
+            phi_f26 = 0.0f;
+        }
+
+        phi_f26 = CLAMP_MAX(phi_f26, 1.0f);
+
+        if (!(globalCtx->cameraPtrs[0]->flags2 & 0x100) || (phi_f26 == 0.0f)) {
+            return;
+        }
+    }
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    for (i = 0; i < this->unk_114C; i++) {
+        spBC.x = this->unk_14C[i].unk_00 + this->unk_14C[i].unk_0C;
+        spBC.y = this->unk_14C[i].unk_04 + this->unk_14C[i].unk_10;
+        spBC.z = this->unk_14C[i].unk_08 + this->unk_14C[i].unk_14;
+        func_80169474(globalCtx, &spBC, &spB0);
+
+        if ((0.0f <= spB0.x) && (spB0.x < 320.0f) && (0.0f <= spB0.y) && (spB0.y < 240.0f)) {
```

Comment:
```suggestion
        if ((spB0.x >= 0.0f) && (spB0.x < 320.0f) && (spB0.y >= 0.0f) && (spB0.y < 240.0f)) {
```
Having literals on the left side of a comparison looks weird imo